test/common/operation.go

```diff
@@ -0,0 +1,221 @@
+}
+
+// SendFakeEventToChannel will send the given event to the given channel.
+func (client *Client) SendFakeEventToChannel(senderName, channelName string, event *base.CloudEvent) error {
+	namespace := client.Namespace
+	channel, err := client.Eventing.EventingV1alpha1().Channels(namespace).Get(channelName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	return client.sendFakeEventToAddressable(senderName, channel.Status.Address, event)
+}
+
+// SendFakeEventToBroker will send the given event to the given broker.
+func (client *Client) SendFakeEventToBroker(senderName, brokerName string, event *base.CloudEvent) error {
+	namespace := client.Namespace
+	broker, err := client.Eventing.EventingV1alpha1().Brokers(namespace).Get(brokerName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	return client.sendFakeEventToAddressable(senderName, broker.Status.Address, event)
+}
+
+// sendFakeEventToAddressable will create a sender pod, which will send the given event to the given url.
+func (client *Client) sendFakeEventToAddressable(senderName string, addr duckv1alpha1.Addressable, event *base.CloudEvent) error {
+	namespace := client.Namespace
+	url := fmt.Sprintf("http://%s", addr.Hostname)
+	client.Logf("Sending fake CloudEvent")
+	pod := base.EventSenderPod(senderName, url, event)
+	if err := client.CreatePod(pod); err != nil {
+		return err
+	}
+	if err := pkgTest.WaitForPodRunning(client.Kube, senderName, namespace); err != nil {
+		return err
+	}
+	return nil
+}
+
+// WaitForBrokerReady waits until the broker is Ready.
+func (client *Client) WaitForBrokerReady(name string) error {
+	namespace := client.Namespace
+	brokers := client.Eventing.EventingV1alpha1().Brokers(namespace)
+	if err := base.WaitForBrokerState(
+		brokers,
+		base.ResourceReadyChecker(client.Dynamic),
+		name,
+		"BrokerIsReady",
+	); err != nil {
+		return err
+	}
+	return nil
+}
+
+// WaitForBrokersReady waits until all brokers in the namespace are Ready.
+func (client *Client) WaitForBrokersReady() error {
+	namespace := client.Namespace
+	brokers := client.Eventing.EventingV1alpha1().Brokers(namespace)
+	if err := base.WaitForBrokerListState(
+		brokers,
+		base.AllResourcesReadyChecker(client.Dynamic),
+		"BrokersAreReady",
+	); err != nil {
+		return err
+	}
+	return nil
+}
+
+// WaitForTriggerReady waits until the trigger is Ready.
+func (client *Client) WaitForTriggerReady(name string) error {
+	namespace := client.Namespace
+	triggers := client.Eventing.EventingV1alpha1().Triggers(namespace)
+	if err := base.WaitForTriggerState(
+		triggers,
+		base.ResourceReadyChecker(client.Dynamic),
+		name,
+		"TriggerIsReady",
+	); err != nil {
+		return err
+	}
+	return nil
+}
+
+// WaitForTriggersReady waits until all triggers in the namespace are Ready.
+func (client *Client) WaitForTriggersReady() error {
+	namespace := client.Namespace
+	triggers := client.Eventing.EventingV1alpha1().Triggers(namespace)
+	if err := base.WaitForTriggerListState(
```

Comment:
You should just be able to return the result from base.WaitForX without needing to capture the `err` value here.

Reply:
Deleted all `WaitForXXListState` functions, and there is no such issue anymore.